lib/spoom/snapshot.rb

```diff
@@ -0,0 +1,96 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Spoom
+  class Snapshot < Hash
```

Comment:
I believe it is fine here, but in general it is bad practice to subclass from a built-in type.